### DISCUSSION TITLE
Flask integration

### DISCUSSION THREAD

**Discussion by Moosems** [tauri-apps/tauri#6529] [Q&A] [answered]
For my front end I just use html and css and use the Flask Python package for backend, the QuickStart has an html, css, and JavaScript tutorial, is there any way I could implement Flask for Tauri?

**Comment by FabianLars** [comment, marked as answer]
Only in an indirect way. You could bundle your flask app using something like pyinstaller or pyoxidizer and then use tauri's sidecar feature to include it in your app https://tauri.app/v1/guides/building/sidecar/

**Reply by Moosems** [reply]
Thanks for your help!

**Reply by simonhyll** [reply]
I just wanna chime in here that using Flask and Python for a your frontend is something I would heavily discourage and urge you to move to a different setup.

Flask is basically like an Express or Actix web server framework. People have put frontend stuff into it, but that doesn't mean that's what it's best suited for. It's a form of SSR solution and the by leagues and miles best solution for you is to pre-render your frontend using SSG/SPA alternatives. It's technically possible to sidecar. You WILL run into issues. If you use Flask, figure out how to turn that flask frontend into a folder with HTML/CSS/JS files in it, as if you were going to send it to be statically hosted on an Apache of Nginx server

You can use Flask. The question is if you should and if you're going to run into so many issues that you're just better off learning how to do it properly.

**Reply by Moosems** [reply]
Yeah and that brings the big issue which is that I can understand rust, c, c++, python, java, etc but I can't understand JS for the life of me

**Reply by simonhyll** [reply]
I mean Typescript and Rust is surprisingly similar syntax wise, I often prototype in Typescript so that I can easily copy-paste it into Rust and then work on converting it to its Rust equivalent. But yea if you're really intent on sticking to Flask then go for it, just know you're going to have issues and you're effectively removing the entire benefit of using Tauri. The webviews that Tauri uses are the system webviews, so comparing it with something like Electron then Electron offers more consistent frontend performance. Where Tauri shines is in its backend, how it handles interactions with that webview and the fact that the code you're running in the backend is in Rust, and our scope systems, all that good stuff. When you decide to use Flask instead you're literally replacing the entire edge of Tauri with Python because there's no way for your Flask server to communicate with the Tauri backend (without using extremely inefficient roundabout solutions like a local REST API server). If you use Flask you're going to have to handle a lot of the security yourself, things like that with Tauri we're hosting your assets over https, so you're going to have to set up things like self signed certificates yourself in order to just get that base line of security

**Comment by WSH032** [comment]
I came across this discussion while searching through Tauri's issues (I know this discussion is from a year ago, but I couldn't resist answering).
`pytauri` has an example using `uvicorn/FastAPI` [here](https://github.com/pytauri/example-nicegui-app/blob/52f8a1265111b6be6db4814588599e8e3370ffb9/python/nicegui_app/__init__.py#L90-L99), which might be helpful to you. (*Disclaimer, I am the author of pytauri, although I don't know how long I can maintain it*)
